refactor(cdr-pagination): cdr-pagination <script setup> refactor

KenjiCrosland opened the pull request:
Remove example with slotted links. Fix issues with simulated clicks from option select. Add
additional test for select. Add uid module of ids. Fix snapshots. Fix Prev and Next link text.

CDR-2516

wjramos commented:
In it intentional that the version be bumped to 1.1 here?

Otherwise, approved

Commit:
refactor(cdr-pagination): cdr-pagination <script setup> refactor

Remove example with slotted links. Fix issues with simulated clicks from option select. Add
additional test for select. Add uid module of ids. Fix snapshots. Fix Prev and Next link text.

CDR-2516

Commit:
refactor(cdr-pagination): remove defineEmits compiler macro

CDR-2516
